src/RpcClient.ts

```diff
@@ -424,6 +435,146 @@ export class RpcClient {
+      }).compileToV0Message(lookupTables)
+    );
+
+    const rpcResponse = await this.connection.simulateTransaction(testTransaction, {
+      replaceRecentBlockhash: true,
+      sigVerify: false,
+    });
+
+    if (rpcResponse.value.err) {
+      console.error(`Simulation error: ${rpcResponse.value.err}`);
+      return null;
+    }
+
+    return rpcResponse.value.unitsConsumed || null;
+  }
+
+  /**
+   * Poll a transaction to confirm whether it has been confirmed
+   * @param {TransactionSignature} txtSig - The transaction signature
+   * @returns {Promise<TransactionSignature>} - The confirmed transaction signature
+  */
+  async pollTransactionConfirmation(txtSig: TransactionSignature): Promise<TransactionSignature> {
+    // 30 second timeout
+    const timeout = 30000;
+    // 5 second retry interval
+    const interval = 5000;
+    let elapsed = 0;
+
+    return new Promise<TransactionSignature>((resolve, reject) => {
+      const intervalId = setInterval(async () => {
+        elapsed += interval;
+
+        if (elapsed >= timeout) {
+          clearInterval(intervalId);
+          reject(new Error(`Transaction ${txtSig}'s confirmation timed out`));
+        }
+
+        const status = await this.connection.getSignatureStatus(txtSig);
+
+        if (status?.value?.confirmationStatus === "finalized") {
+          clearInterval(intervalId);
+          resolve(txtSig);
+        }
+      }, interval);
+    });
+  }
+
+  /**
+   * Build and send an optimized transaction, and handle its confirmation status
+   * @param {TransactionInstruction[]} instructions - The transaction instructions
+   * @param {Keypair} fromKeypair - The sender's keypair
+   * @param {PriorityLevel} priorityLevel - The priority level for the fee
+   * @returns {Promise<TransactionSignature>} - The transaction signature
+  */
+  async sendSmartTransaction(
+    instructions: TransactionInstruction[],
+    fromKeypair: Keypair,
+    priorityLevel: PriorityLevel = PriorityLevel.HIGH
+  ): Promise<TransactionSignature> {
+    try {
+      const pubKey = fromKeypair.publicKey;
+
+      // Build the initial transaction to estimate the priority fee
+      const transaction = new Transaction().add(...instructions);
+      transaction.recentBlockhash = (await this.connection.getLatestBlockhash()).blockhash;
+      transaction.feePayer = pubKey;
+      transaction.sign(fromKeypair);
+
+      // Serialize the transaction
+      const serializedTransaction = bs58.encode(transaction.serialize());
+
+      // Get the priority fee estimate based on the serialized transaction
+      const priorityFeeResponse = await this.getPriorityFeeEstimate({
+        transaction: serializedTransaction,
+        options: { 
+          priorityLevel, 
+        },
+      });
+
+      // Ensure the priority fee is at least 10k lamports
+      let priorityFee = priorityFeeResponse.priorityFeeEstimate || 0;
+      if (priorityFee < 10000) priorityFee = 10000;
```

Comment:
Liam is gonna add this to the recommended response for the fee API — this shouldn't be hardcoded